### PR TITLE
Port to jbuilder

### PR DESCRIPTION
**Pull request by robhoes**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,41 +1,28 @@
...
-.PHONY: build doc test all install uninstall reinstall clean distclean configure
+# requires odoc
+doc:
+	jbuilder build @doc
 
-# OASIS_STOP
+reindent:
+	ocp-indent --syntax cstruct -i lib/*.mli
+	ocp-indent --syntax cstruct -i lib/*.ml
+	ocp-indent --syntax cstruct -i test/*.ml
```
Comment (gaborigloi) [inline Makefile]: We could just say
```
ocp-indent --inplace **/*.ml*
```
to make sure the command stays correct even if we add/remove ml files.

File: ocaml-netlink.opam [hunk 2]
```diff
@@ -2,14 +2,9 @@ opam-version: "1.2"
...
-build: [make]
-install: [make "install"]
-remove: [
-  [make "uninstall"]
-  ["ocamlfind" "remove" "netlink"]
-]
+build: ["jbuilder" "build" "-p" name "-j" jobs]
 depends: [
-  "ocamlfind"
+  "jbuilder"
```
Comment (mseri) [inline ocaml-netlink.opam]: use `"jbuilder" {build}` here

File: test/jbuild [hunk 3]
```diff
@@ -0,0 +1,9 @@
+(executable
+ ((name test)
+  (libraries (ocaml-netlink))
+  ))
+  
+(alias
+ ((name   runtest)
+  (deps   (test.exe))
+  (action (run ${<} -runner sequential))))
```
Comment (mseri) [inline test/jbuild]: I think you don't need `-runner sequential`, it was not specified in the oasis file

File: ocaml-netlink.opam [hunk 4]
```diff
@@ -2,14 +2,9 @@ opam-version: "1.2"
...
-build: [make]
-install: [make "install"]
-remove: [
-  [make "uninstall"]
-  ["ocamlfind" "remove" "netlink"]
-]
+build: ["jbuilder" "build" "-p" name "-j" jobs]
 depends: [
-  "ocamlfind"
+  "jbuilder {build}"
```
Comment (mseri) [inline ocaml-netlink.opam]: `{build}` goes outside the quotes

File: Makefile [hunk 5]
```diff
@@ -1,41 +1,26 @@
...
+	jbuilder runtest
 
-.PHONY: build doc test all install uninstall reinstall clean distclean configure
+# requires odoc
+doc:
+	jbuilder build @doc
 
-# OASIS_STOP
+reindent:
+	ocp-indent --inplace **/*.ml*
```
Comment (thomassa) [inline Makefile]: This matches `.ml` and `.mli` but also potentially other suffixes like `.ml~`, `.mli~`, `.ml.orig` and `.mldylib` if there might ever be such things...

Comment (gaborigloi) [inline Makefile]: Ooops sorry, this was my recommendation.
Maybe we should split it into two commands in the future:
```
ocp-indent --inplace **/*.ml
ocp-indent --inplace **/*.mli
```

Comment (mseri) [inline Makefile]: There should be non of them. But it can still interfere with `mly`

**Pull request merged**